configs/components/ruby-2.4.1.rb

```diff
@@ -0,0 +1,281 @@
+  #   - 8e88a51a: (RE-5401) Add rbconfig for solaris 11 sparc
+  #   - 8f10f5f8: (RE-5400) Roll rbconfig for solaris 11 back to 2.1.6
+  #   - 741d18b1: (RE-5400) Update ruby for solaris 11 i386
+  #   - d09ed06f: (RE-5290) Update ruby to replace rbconfig for all solaris
+  #   - bba35c1e: (RE-5290) Update ruby for a cross-compile on solaris 10
+  rbconfig_info = {
+    'powerpc-ibm-aix5.3.0.0' => {
+      :sum => "5c445630ac67ddf61d16eb47a4ff955b",
+      :target_double => "powerpc-aix5.3.0.0",
+    },
+    'powerpc-ibm-aix6.1.0.0' => {
+      :sum => "82ee3719187cba9bbf6f4b7088b52305",
+      :target_double => "powerpc-aix6.1.0.0",
+    },
+    'powerpc-ibm-aix7.1.0.0' => {
+      :sum => "960fb03d7818fec612f3be598c6964d2",
+      :target_double => "powerpc-aix7.1.0.0",
+     },
+    'powerpc-linux-gnu' => {
+      :sum => "763f316f8f43878d1f3bd5aa6bbe36e8",
+      :target_double => "powerpc-linux",
+    },
+    'powerpc64le-linux-gnu' => {
+      :sum => "c76357879d97fea04bd1d4df099acdec",
+      :target_double => "powerpc64le-linux",
+    },
+    's390x-linux-gnu' => {
+      :sum => "dc6341fff1d00b3ba22dc1b9e6d5532f",
+      :target_double => "s390x-linux",
+    },
+    'i386-pc-solaris2.10' => {
+      :sum => "9078034711ef1b047dcb7416134c55ae",
+      :target_double => 'i386-solaris2.10',
+    },
+    'sparc-sun-solaris2.10' => {
+      :sum => "3fbceb4f70e086a6df52c206ca17211b",
+      :target_double => 'sparc-solaris2.10',
+    },
+    'i386-pc-solaris2.11' => {
+      :sum => "224822682a570aceec965c75ca00d882",
+      :target_double => 'i386-solaris2.11',
+    },
+    'sparc-sun-solaris2.11' => {
+      :sum => "f2a40c4bff028dffc880a40b2806a361",
+      :target_double => 'sparc-solaris2.11',
+    },
+    'arm-linux-gnueabihf' => {
+      :target_double => 'arm-linux-eabihf'
+    },
+    'arm-linux-gnueabi' => {
+      :target_double => 'arm-linux-eabi'
+    },
+    'x86_64-w64-mingw32' => {
+      :sum => "fe5656cd5fcba0a63b18857275e03808",
+      :target_double => 'x64-mingw32',
+    },
+    'i686-w64-mingw32' => {
+      :sum => "795fd622cca6459146cfb226c74ba058",
+      :target_double => 'i386-mingw32',
+    },
+  }
+
+  special_flags = " --prefix=#{settings[:prefix]} --with-opt-dir=#{settings[:prefix]} "
+
+
+  if platform.is_aix?
+    pkg.apply_patch "#{base}/aix_ruby_2.1_libpath_with_opt_dir.patch"
+    pkg.apply_patch "#{base}/aix_ruby_2.1_fix_make_test_failure.patch"
+    pkg.environment "CC" => "/opt/pl-build-tools/bin/gcc"
+    pkg.environment "LDFLAGS" => settings[:ldflags]
+    pkg.build_requires "libedit"
+    pkg.build_requires "runtime"
+
+    # This normalizes the build string to something like AIX 7.1.0.0 rather
+    # than AIX 7.1.0.2 or something
+    special_flags += " --build=#{settings[:platform_triple]} "
+  end
+
+  if platform.is_windows?
+    pkg.apply_patch "#{base}/windows_fixup_generated_batch_files.patch"
+    pkg.apply_patch "#{base}/update_rbinstall_for_windows.patch"
```

Comment:
We should really include ticket numbers in the filenames to help in looking these up - but I was able to do a little archaeology:

* [windows_fixup_generated_batch_files.patc](https://github.com/puppetlabs/puppet-agent/blob/master/resources/patches/ruby_219/windows_fixup_generated_batch_files.patch) is for [PUP-4077](https://tickets.puppetlabs.com/browse/PUP-4077) - this relates to an old issue around running `ci:test:git` - and since we're still cloning `puppet-win32-ruby` for `ci:test:git` ... I believe this patch should stay.  Longer term we should build / ship our custom Ruby to an internal Nuget Feed, at which point this need might go away /cc @McdonaldSeanp 
* [update_rbinstall_for_windows.patch](https://github.com/puppetlabs/puppet-agent/blob/master/resources/patches/ruby_219/update_rbinstall_for_windows.patch) doesn't have a ticket unfortunately, but it comments out a large block of code similar to the code above.  I'm not sure why we're doing this but maybe @McdonaldSeanp knows as he originated the patch...